sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/joinTypes.scala

```diff
@@ -47,4 +48,9 @@ case object RightOuter extends JoinType
 
 case object FullOuter extends JoinType
 
-case object LeftSemi extends JoinType
+abstract class LeftSemiJoin extends JoinType
```

Comment:
Well, actually in https://en.wikipedia.org/wiki/Relational_algebra, `anti-join` also sometimes call `anti-semijoin`, that's the original idea why I make the LeftSemiJoin as an abstract class with 2 objects `LeftSemi` & `LeftAnti`.